safe/gui/tools/batch/batch_dialog.py

```diff
@@ -506,18 +504,13 @@ def run_task(self, task_item, status_item, count=0, index=''):
 
                         # generate map report and impact report
                         try:
-                            # this line is to save the impact report in default
-                            # InaSAFE directory.
-                            generate_impact_report(impact_function, self.iface)
-                            generate_impact_map_report(
-                                impact_function,
-                                self.iface)
+                            error_code, message = generate_report(
+                                impact_function, self.iface)
+
                             # this line is to save the report in user specified
                             # directory.
                             self.generate_pdf_report(
-                                impact_function,
-                                self.iface,
-                                group_name)
+                                impact_function, self.iface, group_name)
                         except:
```

Comment:
May be set the expected exception here

Reply:
I guess it's `Exception` because we need to catch all exceptions. I'm fine for that.